cpp/include/raft/distance/detail/distance_ops/l2_exp.cuh

```diff
@@ -102,7 +102,10 @@ struct l2_exp_distance_op {
     }
   }
 
-  l2_exp_cutlass_op<DataT, AccT> get_cutlass_op() { return l2_exp_cutlass_op<DataT, AccT>(sqrt); }
+  constexpr l2_exp_cutlass_op<DataT, AccT> get_cutlass_op()
```

Comment:
On further thoughts, since this acts as a "create" function that doesn't rely on state of the object - it should be `static constexpr`. That will probably convert my question below to a compiler error.

Reply:
It does rely on the state of the object: the `sqrt` comes from the object. Is there a way that I can document this better?

Reply:
Hmmm I understand it now. No, it's fine as is - my confusion came because I just looked at the diff but looking at the whole file is enough context. Thank you!